R/slice.R

```diff
@@ -174,62 +174,129 @@ vec_assign_fallback <- function(x, i, value) {
 #' @export
 vec_as_index <- function(i, n, names = NULL) {
```

Comment:
Not related to this PR, but coming back to it after a few days and I can't remember which way `as_index` and `as_position` are mapped to `[[` and `[`. Maybe it would better to use "extract" and "subset" like in the error messages?

Reply:
I like giving names to scalar types, like logical is a vector of boolean, integer is a vector of whole numbers, double is a vector of numbers, data frames are a vector of rows or of columns, lists are a vector of objects.

And then index vectors contain positions. Does this make sense?

Reply:
Or more precisely they contain `scalar_type` | `NA`.